pkg/controller/operators/olm/operator.go

```diff
@@ -1299,20 +1301,56 @@ func (a *Operator) syncOLMConfig(obj interface{}) (syncError error) {
 			return err
 		}
 
-		// Filter to unique copies
-		uniqueCopiedCSVs := map[string]struct{}{}
+		copiedCSVNamespaces := map[string]map[string]struct{}{}
 		for _, copiedCSV := range copiedCSVs {
-			uniqueCopiedCSVs[copiedCSV.GetName()] = struct{}{}
+			if _, ok := copiedCSVNamespaces[copiedCSV.GetName()]; !ok {
+				copiedCSVNamespaces[copiedCSV.GetName()] = map[string]struct{}{}
+			}
+			copiedCSVNamespaces[copiedCSV.GetName()][copiedCSV.GetNamespace()] = struct{}{}
```

Comment:
nit: This gets a little hard to read due to the nature of the data-structures and looping we're doing here. A comment explaining why we're doing this would be a nice addition for future contributors.

Reply:
Agreed.